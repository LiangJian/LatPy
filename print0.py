def print0(rank, *args):
  if rank == 0:
    print(args)
