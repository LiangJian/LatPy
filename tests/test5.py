#from mpi4py import MPI
import timeit
import numpy as np
import cupy as cp
import time
from print0 import print0
import sys
import torch
import heat as ht

rank = ht.get_comm().rank
comm = ht.get_comm()
#comm = MPI.COMM_WORLD
#rank = comm.Get_rank()
#ngpu = comm.Get_size()

size = 1000 * int(sys.argv[1])

a = ht.random.rand(size, size, dtype=ht.float64, split=0, device='gpu')
ht.save(a, 'data.h5', 'DATA', mode='w')
comm.Barrier()

print0(rank, a.split)
print0(rank, type(a.larray), a.larray.shape)
print0(rank, a.gshape, a.lshape, a.split)
print(rank, a.larray.dtype, a.larray.device, a.larray.layout )
a.larray = torch.from_numpy(np.random.rand(1000, 1000*int(sys.argv[1]))).cuda(rank)
print0(rank, a.gshape, a.lshape, a.split)
print(rank, a.larray.dtype, a.larray.device, a.larray.layout )

t1 = time.time()
if rank == 0:
  b = ht.load('data.h5', dataset='DATA')
t2 = time.time()
comm.Barrier()
print0(rank, "the loading time is {:.2f} s for a {:.2f} MB matrix.".format((t2-t1),size*size/1024./1024.))

if rank == 0:
  print(rank, b.split)

