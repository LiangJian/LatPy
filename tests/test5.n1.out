0 
<class 'torch.Tensor'> torch.Size([1000, 1000]) 
(1000, 1000) (1000, 1000) 0 
0 torch.float64 cuda:0 torch.strided
(1000, 1000) (1000, 1000) 0 
0 torch.float64 cuda:0 torch.strided
the loading time is 0.04 s for a 0.95 MB matrix. 
0 None
